pkg/compiler/backend/simpleimg.go

```diff
@@ -37,24 +39,51 @@ func (*SimpleImg) BuildImage(opts compiler.CompilerBackendOptions) error {
 	name := opts.ImageName
 	path := opts.SourcePath
 	context := opts.Context
+	writer := NewBackendWriter(!config.LuetCfg.GetGeneral().ShowBuildOutput)
 
 	if context == "" {
 		context = "."
 	}
 	dockerfileName := opts.DockerFileName
 
 	buildarg := []string{"build", "-f", dockerfileName, "-t", name, context}
-	Spinner(22)
-	defer SpinnerStop()
+
+	if config.LuetCfg.GetGeneral().ShowBuildOutput {
+		// We have realtime output from command. Quiet spinner.
+		SpinnerStop()
+	}
+
 	Info(":tea: Building image " + name)
 	cmd := exec.Command("img", buildarg...)
+	cmd.Stdout = writer
+	cmd.Stdout = writer
 	cmd.Dir = path
-	out, err := cmd.CombinedOutput()
 
+	err := cmd.Start()
 	if err != nil {
-		return errors.Wrap(err, "Failed building image: "+string(out))
+		return errors.Wrap(err, "Failed starting build")
 	}
+
+	err = cmd.Wait()
+	if err != nil {
+		return errors.Wrap(err, "Failed waiting for building command")
+	}
+
+	res := cmd.ProcessState.ExitCode()
+	if res != 0 {
+		return errors.Wrap(err,
+			fmt.Sprintf("Failed building image (exiting with %d)", res),
+		)
+	}
+
 	Info(":tea: Building image " + name + " done")
+
+	if config.LuetCfg.GetGeneral().ShowBuildOutput {
+		Spinner(22)
```

Comment:
Are we sure we want to show spinner when outputting logs ?

Reply:
This set previous status after that the build is done

Reply:
See comment above :up: